### ISSUE TITLE
why lettuce client keep reconnecting

### ISSUE THREAD

**Issue by lujia-zhang** [lettuce-io/lettuce-core#861]
We're using connection pooling with maxTotal set to 30. I found in the logs the are lots of reconnection events happened and would like to know why the client keep reconnecting to our redis cluster.

Logs:
```
2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-7] INFO  - Reconnecting, last destination was /192.168.10.89:7004
2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-6] INFO  - Reconnecting, last destination was /192.168.10.89:7000
2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-5] INFO  - Reconnecting, last destination was /192.168.10.89:7000
2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-14] INFO  - Reconnected to 192.168.10.89:7000
2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-13] INFO  - Reconnected to 192.168.10.89:7004
2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-15] INFO  - Reconnected to 192.168.10.89:7000
2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-8] INFO  - Reconnecting, last destination was /192.168.10.11:7000
2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-13] INFO  - Reconnecting, last destination was /192.168.10.159:7003
2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-12] INFO  - Reconnecting, last destination was /192.168.10.159:7000
2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-11] INFO  - Reconnecting, last destination was /192.168.10.203:7005
2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-9] INFO  - Reconnecting, last destination was /192.168.10.203:7004
2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-10] INFO  - Reconnecting, last destination was /192.168.10.203:7005
2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-16] INFO  - Reconnected to 192.168.10.11:7000
2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-1] INFO  - Reconnected to 192.168.10.159:7003
2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-2] INFO  - Reconnected to 192.168.10.159:7000
2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-3] INFO  - Reconnected to 192.168.10.203:7005
2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-4] INFO  - Reconnected to 192.168.10.203:7004
2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-5] INFO  - Reconnected to 192.168.10.203:7005
2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-14] INFO  - Reconnecting, last destination was /192.168.10.11:7002
2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-2] INFO  - Reconnecting, last destination was /192.168.10.203:7002
2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-1] INFO  - Reconnecting, last destination was /192.168.10.89:7004
2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-15] INFO  - Reconnecting, last destination was /192.168.10.89:7003
2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-16] INFO  - Reconnecting, last destination was /192.168.10.203:7000
2018-09-20 10:46:57,302 [lettuce-nioEventLoop-6-6] INFO  - Reconnected to 192.168.10.11:7002
2018-09-20 10:46:57,302 [lettuce-nioEventLoop-6-10] INFO  - Reconnected to 192.168.10.203:7000
2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-8] INFO  - Reconnected to 192.168.10.89:7004
2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-7] INFO  - Reconnected to 192.168.10.203:7002
2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-9] INFO  - Reconnected to 192.168.10.89:7003
2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-3] INFO  - Reconnecting, last destination was /192.168.10.89:7000
2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-4] INFO  - Reconnecting, last destination was /192.168.10.89:7000
2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-5] INFO  - Reconnecting, last destination was /192.168.10.11:7000
2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-11] INFO  - Reconnected to 192.168.10.89:7000
2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-12] INFO  - Reconnected to 192.168.10.89:7000
2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-13] INFO  - Reconnected to 192.168.10.11:7000
2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-6] INFO  - Reconnecting, last destination was /192.168.10.11:7000
2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-8] INFO  - Reconnecting, last destination was /192.168.10.89:7003
2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-7] INFO  - Reconnecting, last destination was /192.168.10.11:7002
2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-16] INFO  - Reconnected to 192.168.10.11:7002
2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-14] INFO  - Reconnected to 192.168.10.11:7000
2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-15] INFO  - Reconnected to 192.168.10.89:7003
2018-09-20 10:46:57,602 [lettuce-eventExecutorLoop-4-10] INFO  - Reconnecting, last destination was /192.168.10.159:7000
2018-09-20 10:46:57,602 [lettuce-eventExecutorLoop-4-9] INFO  - Reconnecting, last destination was /192.168.10.11:7002
2018-09-20 10:46:57,603 [lettuce-nioEventLoop-6-1] INFO  - Reconnected to 192.168.10.159:7000
2018-09-20 10:46:57,603 [lettuce-nioEventLoop-6-2] INFO  - Reconnected to 192.168.10.11:7002
2018-09-20 10:46:57,703 [lettuce-eventExecutorLoop-4-11] INFO  - Reconnecting, last destination was /192.168.10.203:7005
2018-09-20 10:46:57,703 [lettuce-nioEventLoop-6-3] INFO  - Reconnected to 192.168.10.203:7005
2018-09-20 10:46:57,803 [lettuce-eventExecutorLoop-4-12] INFO  - Reconnecting, last destination was /192.168.10.11:7002
2018-09-20 10:46:57,803 [lettuce-nioEventLoop-6-4] INFO  - Reconnected to 192.168.10.11:7002```

**Comment by mp911de** [comment]
What is the client limit in Redis and did you configure disconnect after idle time in Redis?

**Comment by lujia-zhang** [comment]
Thanks for your response!

> What is the client limit in Redis and did you configure disconnect after idle time in Redis?
>

I use connection pooling with maxTotal set to 30. I do use adaptive topology refresh configs:
```ClusterTopologyRefreshOptions topologyRefreshOptions = ClusterTopologyRefreshOptions.builder()
        .enableAllAdaptiveRefreshTriggers()
        .adaptiveRefreshTriggersTimeout(Duration.ofSeconds(adaptiveRefreshTimeout))
        .build();
```
How to config disconnect after idle time in Redis? I don't find any wiki about it.

**Comment by mp911de** [comment]
Take a look at http://download.redis.io/redis-stable/redis.conf, specifically lines

```
# Close the connection after a client is idle for N seconds (0 to disable)
timeout 0
```

and 

```
# Once the limit is reached Redis will close all the new connections sending
# an error 'max number of clients reached'.
#
# maxclients 10000
```

**Comment by lujia-zhang** [comment]
@mp911de Thanks for the info. that's very helpful. I just checked our config and if I understand correctly,
since we set `timeout 300`, the reconnect will happens every 30 seconds when there are no request using the connection?

**Comment by mp911de** [comment]
> the reconnect will happens every 300 seconds when there are no request using the connection?

This exactly hits the point. Lettuce is designed to maintain long-lived connections. Summarizing this issue, Lettuce behaves as designed as it reconnects after Redis disconnects idle connections regularly.

May I close the issue?

**Comment by gppwin** [comment]
hey, I also encountered this problem.  project is spring-boot 2.0，config the lettuce，
Can you tell me why not let the reconnect log?

**Comment by mp911de** [comment]
> Can you tell me why not let the reconnect log?

Can you rephrase your question? I don't understand this question.

**Comment by gppwin** [comment]
> > Can you tell me why not let the reconnect log?
> 
> Can you rephrase your question? I don't understand this question.

Reconnect every few minutes

**Comment by gppwin** [comment]
> > Can you tell me why not let the reconnect log?
> 
> Can you rephrase your question? I don't understand this question.

Can I set it? Do not print

**Comment by gppwin** [comment]
this is my Configuration
redis:
    password:
    timeout: 50s
    cluster:
      nodes:
        - 172.19.240.42:6379
        - 172.19.240.42:6380
        - 172.19.240.43:6379
        - 172.19.240.43:6380
        - 172.19.240.44:6379
        - 172.19.240.44:6380
      max-redirects: 6
    lettuce:
      pool:
        max-active: 8
        max-wait: -1ms
        max-idle: 8
        min-idle: 0

**Comment by gppwin** [comment]
> We're using connection pooling with maxTotal set to 30. I found in the logs the are lots of reconnection events happened and would like to know why the client keep reconnecting to our redis cluster.
> 
> Logs:
> 
> ```
> 2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-7] INFO  - Reconnecting, last destination was /192.168.10.89:7004
> 2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-6] INFO  - Reconnecting, last destination was /192.168.10.89:7000
> 2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-5] INFO  - Reconnecting, last destination was /192.168.10.89:7000
> 2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-14] INFO  - Reconnected to 192.168.10.89:7000
> 2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-13] INFO  - Reconnected to 192.168.10.89:7004
> 2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-15] INFO  - Reconnected to 192.168.10.89:7000
> 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-8] INFO  - Reconnecting, last destination was /192.168.10.11:7000
> 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-13] INFO  - Reconnecting, last destination was /192.168.10.159:7003
> 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-12] INFO  - Reconnecting, last destination was /192.168.10.159:7000
> 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-11] INFO  - Reconnecting, last destination was /192.168.10.203:7005
> 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-9] INFO  - Reconnecting, last destination was /192.168.10.203:7004
> 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-10] INFO  - Reconnecting, last destination was /192.168.10.203:7005
> 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-16] INFO  - Reconnected to 192.168.10.11:7000
> 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-1] INFO  - Reconnected to 192.168.10.159:7003
> 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-2] INFO  - Reconnected to 192.168.10.159:7000
> 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-3] INFO  - Reconnected to 192.168.10.203:7005
> 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-4] INFO  - Reconnected to 192.168.10.203:7004
> 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-5] INFO  - Reconnected to 192.168.10.203:7005
> 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-14] INFO  - Reconnecting, last destination was /192.168.10.11:7002
> 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-2] INFO  - Reconnecting, last destination was /192.168.10.203:7002
> 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-1] INFO  - Reconnecting, last destination was /192.168.10.89:7004
> 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-15] INFO  - Reconnecting, last destination was /192.168.10.89:7003
> 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-16] INFO  - Reconnecting, last destination was /192.168.10.203:7000
> 2018-09-20 10:46:57,302 [lettuce-nioEventLoop-6-6] INFO  - Reconnected to 192.168.10.11:7002
> 2018-09-20 10:46:57,302 [lettuce-nioEventLoop-6-10] INFO  - Reconnected to 192.168.10.203:7000
> 2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-8] INFO  - Reconnected to 192.168.10.89:7004
> 2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-7] INFO  - Reconnected to 192.168.10.203:7002
> 2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-9] INFO  - Reconnected to 192.168.10.89:7003
> 2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-3] INFO  - Reconnecting, last destination was /192.168.10.89:7000
> 2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-4] INFO  - Reconnecting, last destination was /192.168.10.89:7000
> 2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-5] INFO  - Reconnecting, last destination was /192.168.10.11:7000
> 2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-11] INFO  - Reconnected to 192.168.10.89:7000
> 2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-12] INFO  - Reconnected to 192.168.10.89:7000
> 2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-13] INFO  - Reconnected to 192.168.10.11:7000
> 2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-6] INFO  - Reconnecting, last destination was /192.168.10.11:7000
> 2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-8] INFO  - Reconnecting, last destination was /192.168.10.89:7003
> 2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-7] INFO  - Reconnecting, last destination was /192.168.10.11:7002
> 2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-16] INFO  - Reconnected to 192.168.10.11:7002
> 2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-14] INFO  - Reconnected to 192.168.10.11:7000
> 2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-15] INFO  - Reconnected to 192.168.10.89:7003
> 2018-09-20 10:46:57,602 [lettuce-eventExecutorLoop-4-10] INFO  - Reconnecting, last destination was /192.168.10.159:7000
> 2018-09-20 10:46:57,602 [lettuce-eventExecutorLoop-4-9] INFO  - Reconnecting, last destination was /192.168.10.11:7002
> 2018-09-20 10:46:57,603 [lettuce-nioEventLoop-6-1] INFO  - Reconnected to 192.168.10.159:7000
> 2018-09-20 10:46:57,603 [lettuce-nioEventLoop-6-2] INFO  - Reconnected to 192.168.10.11:7002
> 2018-09-20 10:46:57,703 [lettuce-eventExecutorLoop-4-11] INFO  - Reconnecting, last destination was /192.168.10.203:7005
> 2018-09-20 10:46:57,703 [lettuce-nioEventLoop-6-3] INFO  - Reconnected to 192.168.10.203:7005
> 2018-09-20 10:46:57,803 [lettuce-eventExecutorLoop-4-12] INFO  - Reconnecting, last destination was /192.168.10.11:7002
> 2018-09-20 10:46:57,803 [lettuce-nioEventLoop-6-4] INFO  - Reconnected to 192.168.10.11:7002```
> ```

你最后是怎么配置的，大兄弟

**Comment by greenHard** [comment]
> > We're using connection pooling with maxTotal set to 30. I found in the logs the are lots of reconnection events happened and would like to know why the client keep reconnecting to our redis cluster.
> > Logs:
> > ```
> > 2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-7] INFO  - Reconnecting, last destination was /192.168.10.89:7004
> > 2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-6] INFO  - Reconnecting, last destination was /192.168.10.89:7000
> > 2018-09-20 10:46:57,103 [lettuce-eventExecutorLoop-4-5] INFO  - Reconnecting, last destination was /192.168.10.89:7000
> > 2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-14] INFO  - Reconnected to 192.168.10.89:7000
> > 2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-13] INFO  - Reconnected to 192.168.10.89:7004
> > 2018-09-20 10:46:57,103 [lettuce-nioEventLoop-6-15] INFO  - Reconnected to 192.168.10.89:7000
> > 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-8] INFO  - Reconnecting, last destination was /192.168.10.11:7000
> > 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-13] INFO  - Reconnecting, last destination was /192.168.10.159:7003
> > 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-12] INFO  - Reconnecting, last destination was /192.168.10.159:7000
> > 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-11] INFO  - Reconnecting, last destination was /192.168.10.203:7005
> > 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-9] INFO  - Reconnecting, last destination was /192.168.10.203:7004
> > 2018-09-20 10:46:57,203 [lettuce-eventExecutorLoop-4-10] INFO  - Reconnecting, last destination was /192.168.10.203:7005
> > 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-16] INFO  - Reconnected to 192.168.10.11:7000
> > 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-1] INFO  - Reconnected to 192.168.10.159:7003
> > 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-2] INFO  - Reconnected to 192.168.10.159:7000
> > 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-3] INFO  - Reconnected to 192.168.10.203:7005
> > 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-4] INFO  - Reconnected to 192.168.10.203:7004
> > 2018-09-20 10:46:57,203 [lettuce-nioEventLoop-6-5] INFO  - Reconnected to 192.168.10.203:7005
> > 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-14] INFO  - Reconnecting, last destination was /192.168.10.11:7002
> > 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-2] INFO  - Reconnecting, last destination was /192.168.10.203:7002
> > 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-1] INFO  - Reconnecting, last destination was /192.168.10.89:7004
> > 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-15] INFO  - Reconnecting, last destination was /192.168.10.89:7003
> > 2018-09-20 10:46:57,302 [lettuce-eventExecutorLoop-4-16] INFO  - Reconnecting, last destination was /192.168.10.203:7000
> > 2018-09-20 10:46:57,302 [lettuce-nioEventLoop-6-6] INFO  - Reconnected to 192.168.10.11:7002
> > 2018-09-20 10:46:57,302 [lettuce-nioEventLoop-6-10] INFO  - Reconnected to 192.168.10.203:7000
> > 2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-8] INFO  - Reconnected to 192.168.10.89:7004
> > 2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-7] INFO  - Reconnected to 192.168.10.203:7002
> > 2018-09-20 10:46:57,303 [lettuce-nioEventLoop-6-9] INFO  - Reconnected to 192.168.10.89:7003
> > 2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-3] INFO  - Reconnecting, last destination was /192.168.10.89:7000
> > 2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-4] INFO  - Reconnecting, last destination was /192.168.10.89:7000
> > 2018-09-20 10:46:57,402 [lettuce-eventExecutorLoop-4-5] INFO  - Reconnecting, last destination was /192.168.10.11:7000
> > 2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-11] INFO  - Reconnected to 192.168.10.89:7000
> > 2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-12] INFO  - Reconnected to 192.168.10.89:7000
> > 2018-09-20 10:46:57,403 [lettuce-nioEventLoop-6-13] INFO  - Reconnected to 192.168.10.11:7000
> > 2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-6] INFO  - Reconnecting, last destination was /192.168.10.11:7000
> > 2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-8] INFO  - Reconnecting, last destination was /192.168.10.89:7003
> > 2018-09-20 10:46:57,502 [lettuce-eventExecutorLoop-4-7] INFO  - Reconnecting, last destination was /192.168.10.11:7002
> > 2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-16] INFO  - Reconnected to 192.168.10.11:7002
> > 2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-14] INFO  - Reconnected to 192.168.10.11:7000
> > 2018-09-20 10:46:57,503 [lettuce-nioEventLoop-6-15] INFO  - Reconnected to 192.168.10.89:7003
> > 2018-09-20 10:46:57,602 [lettuce-eventExecutorLoop-4-10] INFO  - Reconnecting, last destination was /192.168.10.159:7000
> > 2018-09-20 10:46:57,602 [lettuce-eventExecutorLoop-4-9] INFO  - Reconnecting, last destination was /192.168.10.11:7002
> > 2018-09-20 10:46:57,603 [lettuce-nioEventLoop-6-1] INFO  - Reconnected to 192.168.10.159:7000
> > 2018-09-20 10:46:57,603 [lettuce-nioEventLoop-6-2] INFO  - Reconnected to 192.168.10.11:7002
> > 2018-09-20 10:46:57,703 [lettuce-eventExecutorLoop-4-11] INFO  - Reconnecting, last destination was /192.168.10.203:7005
> > 2018-09-20 10:46:57,703 [lettuce-nioEventLoop-6-3] INFO  - Reconnected to 192.168.10.203:7005
> > 2018-09-20 10:46:57,803 [lettuce-eventExecutorLoop-4-12] INFO  - Reconnecting, last destination was /192.168.10.11:7002
> > 2018-09-20 10:46:57,803 [lettuce-nioEventLoop-6-4] INFO  - Reconnected to 192.168.10.11:7002```
> > ```
> 
> 你最后是怎么配置的，大兄弟
修改redis.conf 里面的客户端端开的时间为300秒。这样就能控制5分钟不连接才会去重试。可以根据你业务需求来修改

**Comment by perlun** [comment]
@gppwin @greenHard Please do not use Chinese (or other foreign languages) in a GitHub issue for a global FOSS project; it makes it very hard for the rest of us to participate in the discussion.

**Comment by yoeb** [comment]
聊出母语~~

**Comment by malaya1982** [comment]
那你为什么学java？ Java在编程时不支持您的母语

**Comment by verineliu** [comment]
1

**Comment by leeychee** [comment]
@mp911de So what's the best practice for `timeout` config when we are using lettuce? What factors need to be considered?

FYI:
We have two clusters, both has 6 main + 6 replica. 
cluster 1: `timeout` = 0, works fine
cluster 2:  `timeout` = 60s, sometimes got `Currently not connected. Commands are rejected.`
